src/ITensorNetworks/MPScalculus/contract.jl

```diff
@@ -0,0 +1,54 @@
+using ChainRulesCore
+
+struct MPSTensor
+  mps::MPS
+  cutoff::Float64
+  maxdim::Integer
+end
+
+AbstractTensor = Union{ITensor,MPSTensor}
+
+function ITensors.contract(mps1::MPS, mps2::MPS; cutoff=1e-15, maxdim=1000)
+  ## TODO: modify this function based on https://arxiv.org/pdf/1912.03014.pdf
+  tensor = contract(vcat(mps1.data, mps2.data)...)
```

Comment:
You could simplify this to `contract(mps1..., mps2...)`, or otherwise use `contract(vcat(collect(mps1), collect(mps2))...)`. Generally I would discourage accessing the `data` field directly (I consider it to be an internal implementation detail), `MPS` should work like a `Vector{ITensor}` but if there are operations that are missing then let me know and we can add them.

Reply:
done